cpp/src/parquet/arrow/reader.cc

```diff
@@ -174,10 +174,10 @@ class FileReaderImpl : public FileReader {
   Status BoundsCheck(const std::vector<int>& row_groups,
                      const std::vector<int>& column_indices) {
     for (int i : row_groups) {
-      return BoundsCheckRowGroup(i);
+      RETURN_NOT_OK(BoundsCheckRowGroup(i));
```

Comment:
ping @bkietz I'm happy to revert this commit and have another pr if preferred.

Reply:
No need to wait. That's an embarrassing bug, thanks for catching it